src/Kestrel.Core/KestrelServerOptions.cs

```diff
@@ -105,10 +115,50 @@ public void ConfigureHttpsDefaults(Action<HttpsConnectionAdapterOptions> configu
 
         internal void ApplyHttpsDefaults(HttpsConnectionAdapterOptions httpsOptions)
         {
-            httpsOptions.ServerCertificate = DefaultCertificate;
             HttpsDefaults(httpsOptions);
         }
 
+        internal void ApplyDefaultCert(HttpsConnectionAdapterOptions httpsOptions)
```

Comment:
Nit: Rename to TryApplyDefaultCert of EnsureDefaultCert.

Reply:
I thought about that but you'd normally use Try when you were going to return the status and act differently. Here it just no-ops.

Reply:
EnsureServerCert it is then?